### ISSUE TITLE
Allow to ENABLE/DISABLE any step in button

### ISSUE THREAD

**Issue by Suenee** [bitfocus/companion#3154]
### Is this a feature relevant to companion itself, and not a module?

- [x] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the feature

Assumption:
* I have a button that has more than one step.

For each step, I can set whether it is ENABLED/DISABLED.

If it is ENABLED:
The step is executed as always.

If DISABLED:
The step is skipped.

Define a function that allows you to dynamically specify whether a step is executed or not. It is enough to know the address of the button, the number of the step and its desired state.

USECASE:
Dynamically change the behavior and number of button steps. In practice, I have one button that calls up advertising slogans. If I had this feature available, I could dynamically change which ad slogans are shown. If the button has only one step, it cannot be disabled. If the button has more than one step, only "x-1" can be deactivated from the total number of "x" steps.

**Comment by BlakesHeaven** [comment]
I also need this feature.

**Usecase**
We have three PTZ Cams - 1, 2 & 3. 
We have a set of buttons that set PTZ-# to PreView for several preset locations, where # equals whatever is **not** LIVE:
- ie When PTZ-1 is LIVE then 2 or 3 can be set to preview, when 2 is LIVE then 1 or 3 can be set... etc.

The movable Camera numbers are assigned to custom variables Movable_CamA and Movable_CamB.
We have a button to toggle between the two NON-LIVE PTZ Cams; this button has the Movable_CamA variable in Step 1 and Movable_CamB in Step 2

I need to be able to have a third step which is disabled when one of the cams is LIVE, but when none of the cameras are LIVE (when showing slides or video), I need the step to be enabled so that Movable_CamC (for PTZ-3) can be included in the mix of selectable Cameras that are not LIVE.

Thanks.

**Comment by patrichard** [comment]
Yes, I realize this is an old thread, but it's still open. You can do this more or less in 4.x and later. Add a step with no actions in it. Set the button's step progression to be manual. On OTHER buttons, set this button (internal: button: set current step) to that empty step. Now pressing that button will do nothing until you set it back to Step 1 (or whichever step isn't empty). It's a workaround for not having the ability to disable/enable buttons. I use it for supersource layouts and transitions, by "disabling" any button/layout that I shouldn't transition to (because it might look weird, cause issues, etc.). I then also set the "disabled" button text and background to something else, indicating it's "disabled". It takes a few steps, but accomplishes what's needed. I know in later builds like the 5.x there is a 'internal: button: skip step" option, too. I haven't played with that.

**Comment by Julusian** [comment]
Yeah, maybe this is solved now.
Since v4.1, it has been possible to define the current step as an expression, so perhaps combining that expression and a local-variable for storing which steps are disabled, will I suspect make this possible
